### DISCUSSION TITLE
Is installing websockets a requirement?

### DISCUSSION THREAD

**Discussion by zbs** [fastapi/fastapi#6139] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# no code
```

### Description

The documentation asks users to install `websockets` ([link](https://fastapi.tiangolo.com/advanced/websockets/#install-websockets)); however, I don't see any dependency on the `websockets` library in the code. It seems like all `websocket` dependencies are satisfied via `starlette`, which also does not seem to use it.

### Operating System

Linux

### FastAPI Version

0.85.0

### Python Version

3.8.13

**Comment by JarroVGIT** [comment]
On my phone right now but I believe it is a dependency of Uvicorn to deal with websocket connections.

**Comment by iudeen** [comment, marked as answer]
Websockets interface is provided by the ASGI server you use to serve the FastAPI app. In case of the example, we use Uvicorn, and Websockets is an optional requirement of Uvicorn to enable Websocket. (Alternatively, you can use wsproto as well.)

Note: Hypercorn (another ASGI server) uses wsproto

**Comment by zbs** [comment]
Thanks! That answers it.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

Thanks for reporting back and closing the issue 👍
